File: MapApp/app/src/main/java/com/syrnnik/mapapp/Toilet.java
package com.syrnnik.mapapp;

import org.json.JSONException;
import org.json.JSONObject;

public class Toilet {

    private final int id;
    private double lat;
    private double lon;
    private String title;
    private String address;
    private String desc;

    public Toilet(int id, double lat, double lon, String title, String address, String desc) {
        this.id = id;
        this.lat = lat;
        this.lon = lon;
        this.title = title;
        this.address = address;
        this.desc = desc;
    }

    public int getId() {
        return this.id;
    }

    public double getLat() {
        return this.lat;
    }

    public double getLon() {
        return this.lon;
    }

    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setCoords(double lat, double lon) {
        this.lat = lat;
        this.lon = lon;
    }

    public String getAddress() {
        return this.address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getDesc() {
        return this.desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public JSONObject toJSON() throws JSONException {
        JSONObject toilet = new JSONObject();
        toilet.put("id", this.getId());
        toilet.put("lat", this.getLat());
        toilet.put("lon", this.getLon());
        toilet.put("title", this.getTitle());
        toilet.put("address", this.getAddress());
        toilet.put("desc", this.getDesc());
        return toilet;
    }

}

File: API/backend/OLD/Toilet.py
class Toilet:
    
    def __init__(self, id:int, lat:float, lon:float, title:str, address:str, desc:str=""):
        self.id = id
        self.lat = lat
        self.lon = lon
        self.title = title
        self.desc = desc
        self.address = address
        
    def getCoords(self) -> tuple:
        return (self.lat, self.lon)
    
    def setCoords(self, lat, lon):
        self.lat = lat
        self.lon = lon
    
    def getTitle(self) -> str:
        return self.title
    
    def setTitle(self, title):
        self.title = title
        
    def getAddress(self) -> str:
        return self.address
    
    def setAddress(self, address):
        self.address = address
        
    def getDesc(self) -> str:
        return self.desc
    
    def setDesc(self, desc):
        self.desc = desc
        
    def toJSON(self) -> dict:
        toilet = {
            'id': self.id,
            'lat': self.lat,
            'lon': self.lon,
            'title': self.title,
            'address': self.address,
            'desc': self.desc
        }
        return toilet